cmd/nerdctl/compose_build_linux_test.go

```diff
@@ -0,0 +1,65 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/containerd/nerdctl/pkg/testutil"
+)
+
+func TestComposeBuild(t *testing.T) {
+	testutil.RequiresBuild(t)
+	base := testutil.NewBase(t)
+	defer base.Cmd("builder", "prune").Run()
+
+	const dockerComposeYAML = `
```

Comment:
Does this work on macOs as well; curious. Docker does not work for me for different archs for reasons unknown.

Reply:
Seems it's because docker doesn't support multi-arch build. I'm following the compose file [here](https://github.com/containerd/nerdctl/blob/2f70e700d5352c1d326c653e7659c017e9aaaca4/cmd/nerdctl/multi_platform_linux_test.go#L84).

Added a `DockerIncompatible` to the test as well.

Reply:
There is no need to add DockerIncompatible imo. You can build for same arch; same image (or perhaps different) as long as tags are differently generated and have the test be docker compatible.

Reply:
This is a Docker-compatible feature so I don't think a `DockerIncompatible` should be here. You can split two arches' build into two unit tests.

Reply:
I see. removed it, thanks